### PR TITLE
Patch Curl: CVE-2021-22898 and CVE-2021-22901

### PR DESCRIPTION
**Pull request by jslobodzian**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
CVE Fixes for security vulnerabilities in curl

###### Does this affect the toolchain?  
**YES**
NO

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2021-22898
- https://nvd.nist.gov/vuln/detail/CVE-2021-22901

### CODE REVIEW

**Comment by MateuszMalisz** [comment]
Is this based on a correct branch?

**Comment by MateuszMalisz** [comment]
Asking since there are 250+ modified files... :)

File: SPECS/curl/curl.spec [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
-Release:        1%{?dist}
+Release:        2%{?dist}
 License:        MIT
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          System Environment/NetworkingLibraries
 URL:            https://curl.haxx.se
 Source0:        https://curl.haxx.se/download/%{name}-%{version}.tar.gz
+Patch0:         CVE-2021-22898.patch
+#Patch1:         CVE-2021-22901.patch
```
Comment (PawelWMS) [inline SPECS/curl/curl.spec:12]: I think you missed uncommenting this line.

File: toolkit/resources/manifests/package/pkggen_core_x86_64.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 filesystem-1.1-7.cm1.x86_64.rpm
-kernel-headers-5.10.32.1-4.cm1.noarch.rpm
+kernel-headers-5.10.32.1-3.cm1.noarch.rpm
```
Comment (PawelWMS) [inline toolkit/resources/manifests/package/pkggen_core_x86_64.txt:2]: Why are there changes for other packages than `curl`?

**Review by PawelWMS — CHANGES_REQUESTED** [review]
We should address the spec comment and the manifest changes.

File: toolkit/resources/manifests/package/pkggen_core_x86_64.txt (hunk 2 above)
Comment (MateuszMalisz) [reply toolkit/resources/manifests/package/pkggen_core_x86_64.txt:2]: Applies to other files too.

File: SPECS/curl/curl.spec (hunk 1 above)
Comment (jslobodzian) [reply SPECS/curl/curl.spec:12]: yikes...forgot to save it.  good call.

File: toolkit/resources/manifests/package/pkggen_core_x86_64.txt (hunk 2 above)
Comment (jslobodzian) [reply toolkit/resources/manifests/package/pkggen_core_x86_64.txt:2]: needs to be refreshed with 1.0-dev

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**